extensions/panache/hibernate-orm-panache-rest/runtime/src/main/java/io/quarkus/hibernate/orm/panache/rest/PanacheCrudController.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.hibernate.orm.panache.rest;
+
+import javax.ws.rs.core.Response;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
+
+public interface PanacheCrudController<T extends PanacheEntityBase, ID> {
```

Comment:
Why not returing the type `T`/void/List<T>` instead of a `Response` object ?

Reply:
Having a `Response` as a return type was a bit easier for code generation. I was able to reuse the same code bits for different methods (GET, POST...). Since user doesn't see the generated code, I think this shouldn't be an issue.

Reply:
Initially I did have it as `T`

Reply:
> Since user doesn't see the generated code

Will not always be true, see my other comments.

Returing `Response` can also lead to issues for GraalVM as the entity type may not been discovered at native image generation time (should not be an issue with Hibernate as all entities are registered for relfection but for MongoDB with the repository pattern it's still possible).

Reply:
This depends on how we do method override and super-delegation. We already have another discussion open about this.

Reply:
So would this be an issue with the methods that return an entity via a `Response` or for all methods that return `Response`?

Reply:
With the one returning an entity via a `Response`.

If it's too much work, I'm OK with `Response` but reflection is tricky on GraalVM and having the entity return type instead of `Response` guarantee that the type will be correctly registered for reflection.

But as @FroMage said, it depends how we override methods because if the correct type is not used at code generation type this would be useless.

Reply:
I don't see why returning `Response` would be a problem, since we always know what to register for reflection this case (the entity itself).

Reply:
A `Response` is annoying because it's not typed. If you need to do something on the super method's response type (add an entry to a `List` or iterate it, or set a field on the returned `Order` you lose that info).

OTOH it's much more capable as a return type because it allows you to set response headers and status codes.

IMO, if we make it possible to invoke/delegate to the super method (I think we should), and we do that by declaring a new method (to be able to invoke the old one) then it makes sense to make the super method have the type-safe return type (`Order`, say) since we have control over the overriding method's return type and we can choose to make it return an `Order` too or a `Response` if we need extra stuff.